flax/experimental/nnx/nnx/graph_utils.py

```diff
@@ -41,6 +44,45 @@
 
 NODE_TYPES: dict[type, 'NodeImpl[tp.Any, tp.Any, tp.Any]'] = {}
 
+class _HashById(tp.Hashable, tp.Generic[A]):
```

Comment:
Perhaps, add a docstring explaining that this retains a reference for a value to make sure the `id()` doesn't gets recycled for another unrelated object?